src/gui/src/layoutViewer.cpp

```diff
@@ -1961,6 +1961,9 @@ void LayoutViewer::drawBlock(QPainter* painter,
   drawInstanceNames(painter, insts);
 
   drawRows(painter, bounds);
+  if (options_->areAccessPointsVisible()) {
+    drawAccessPoints(gui_painter);
+  }
```

Comment:
This won't work correctly as you are drawing layer based data.  It will be out of order.  I think it is better to handle APs in drawInstanceShapes.  That method also handles the case where you are so zoomed out that the instance is too small to draw - in which case the APs are too.

Reply:
Can you clarify what do you mean by out of order? do you mean it's gonna be drawn on top of each other?

Reply:
You should draw all objects on m1, then all objects on m2, etc.   That way higher layer objects draw over lower objects.

If you do this you will draw all apps on m1, m2 then all other shapes on m1 then m2.  An m1 rect may cover an m2 AP.

Reply:
got it

Reply:
@maliberty The problem now is the access points are not visible when higher layer are drawn on top of them

Reply:
So turn off the higher layer visibility in the display controls.  That's true of all objects.

Reply:
I know that but is that the use case we want. Don't we usually want to see access points with the nets? If I do the suggested solution, we will only see access points on instances with hiding the nets.

Reply:
I think they are only useful when you are looking at the routing around a pin.  I would think it confusing to see APs when looking at higher layers, especially if they come from iterms that aren't even part of the net you are interested in.

If you have an AP on m1 and a via on it from m2 then hopefully you can still see the X with the usual transparency.   Can you include a screenshot of what that looks like?

Reply:
![GUI](https://user-images.githubusercontent.com/56893454/147418502-16709fbb-e312-4f45-a430-cdc73867c881.png)
You can hardly see the access point

Reply:
And that's just with wire shapes from metal1 without showing via1.

Reply:
This example will look exactly the same either way since you are just drawing one layer.  You will draw the AP after the m1 shape in either case.

Is there just one AP?

Reply:
It doesn't look the same. previously I was drawing access points on top of all shapes; at the end of drawBlock. now I draw it with instance shapes. Net shapes are drawn after instance shapes, so on top of them. That's why you can't see the access points. There are actually 6 access points in this picture; 5 of them you can't see.

![GUI](https://user-images.githubusercontent.com/56893454/147468363-82febf4b-1783-4657-8224-75e21d1b1ef7.png)

Reply:
I'm ok with drawing the pa markers separately in the gui.  However you should handle the zoomed out issue in a similar fashion to instance drawing